dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1492,19 +1497,29 @@ def median_abs_deviation(self) -> float:
             np.append([True], np.diff(bin_edges_impose) > 1e-14)
         ]
 
-        bin_counts_impose_pos: float = np.interp(
-            bin_edges_impose, bin_edges_pos, np.cumsum(np.append([0], bin_counts_pos))
+        bin_counts_impose_pos: float = cast(
+            float,
+            np.interp(
+                bin_edges_impose,
+                bin_edges_pos,
+                np.cumsum(np.append([0], bin_counts_pos)),
+            ),
         )
-        bin_counts_impose_neg: float = np.interp(
-            bin_edges_impose, bin_edges_neg, np.cumsum(np.append([0], bin_counts_neg))
+        bin_counts_impose_neg: float = cast(
+            float,
+            np.interp(
+                bin_edges_impose,
+                bin_edges_neg,
+                np.cumsum(np.append([0], bin_counts_neg)),
+            ),
         )
         bin_counts_impose = bin_counts_impose_pos + bin_counts_impose_neg
 
         median_inds = np.abs(bin_counts_impose - 0.5) < 1e-10
         if np.sum(median_inds) > 1:
-            return np.mean(bin_edges_impose[median_inds])
+            return cast(float, np.mean(bin_edges_impose[median_inds]))
```

Comment:
this would be a float64

Reply:
I'm getting a similar issue as before:
```
"median_absolute_deviation": utils.find_diff_of_numbers(
         self.median_abs_deviation, other_profile.median_abs_deviation
 ),
```
```
dataprofiler/profilers/numerical_column_stats.py:381: error: Value of type variable "T" of "find_diff_of_numbers" cannot be "object"
```
Looking into it more, I believe it's because we're trying to assign a `Union` to a `TypeVar` with `bound=Subtractable` and the `Union` itself does not follow the `Subtractable` protocol. So far the only reasonable solution I can think of is doing:
```
"median_absolute_deviation": utils.find_diff_of_numbers(
         cast(float, self.median_abs_deviation), cast(float, other_profile.median_abs_deviation)
 ),
```
which isn't very ideal. I'll look into mypy generics more to see if there's a workaround.

Reply:
Fixed. See https://github.com/capitalone/DataProfiler/pull/703#discussion_r1015985771